docs/config.md

```diff
@@ -648,6 +649,21 @@ Default: `""`
 
 Type: `string` (url)
 
+### `Gateway.FastDirIndexThreshold`
+
+The minimal number of items in a directory that triggers Gateway
+to switch to an inexpensive HTML listing generation.
+
+This allows for listing big directories with tens of thousands of items
+can be returned as soon as the root node of dag-pb directory is
+resolved, without reading any additional metadata from child nodes.
+
+Setting to 0 will enable fast listings for all directories.
+
+Default: `100`
+
+Type: `optionalInteger`
```

Comment:
:pray: Any way to make this sound less awkward?

Reply:
Maybe something like...

"The minimum number of items in a directory before the Gateway switches to a shallow, faster listing which only considers the root-most items."

Reply:
Simplified in https://github.com/ipfs/go-ipfs/pull/8853/commits/736d6de0a16b96343045f69ab4a4cb1865ed2b07